DuckDuckGo/SettingsSubscriptionView.swift

```diff
@@ -244,23 +245,27 @@ struct SettingsSubscriptionView: View {
                         footer: !isSignedIn ? footerLink : nil
                 ) {
 
-                    switch (isSignedIn, hasActiveSubscription, hasNoEntitlements) {
+                    switch (isSignedIn, hasSubscription, hasActiveSubscription, hasAnyEntitlements) {
 
-                        // Signed In, Subscription Expired
-                    case (true, false, _):
+                    // Signed out
+                    case (false, _, _, _):
+                        purchaseSubscriptionView
+
+                    // Signed In, Subscription Missing
+                    case (true, false, _, _):
+                        missingSubscriptionOrEntitlementsView
```

Comment:
This case is newly introduced but we reuse the same view/state as when "Signed in, Subscription Present & Active, Missing Entitlements"